src/javascript/app_2/App/Components/Elements/popconfirm.jsx

```diff
@@ -0,0 +1,94 @@
+import React        from 'react';
+import classNames   from 'classnames';
+import PropTypes    from 'prop-types';
+import { localize } from '../../../../_common/localize';
+
+class PopConfirm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            is_open: false,
+        };
+        this.handleClose        = this.handleClose.bind(this);
+        this.setWrapperRef      = this.setWrapperRef.bind(this);
+        this.handleClickOutside = this.handleClickOutside.bind(this);
+    }
+
+    componentDidMount() {
+        document.addEventListener('mousedown', this.handleClickOutside);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('mousedown', this.handleClickOutside);
+    }
+
+    setWrapperRef(node) {
+        this.wrapper_ref = node;
+    }
+
+    handleClickOutside(event) {
+        if (this.wrapper_ref && !this.wrapper_ref.contains(event.target) && this.state.is_open) {
+            this.setState({ is_open: false });
+        }
+    }
+
+    handleClose() {
+        this.setState({ is_open: false });
```

Comment:
suggestion to use arrow function: ` handleClose = () => {...} ` and remove `this.handleClose.bind(this);` in constructor